megfile/errors.py

```diff
@@ -75,7 +75,13 @@ def full_error_message(error):
 
 
 def client_error_code(error: ClientError) -> str:
-    return error.response.get('Error', {}).get('Code', 'Unknown')  # pytype: disable=attribute-error
+    err = error.response.get('Error', {})  # pytype: disable=attribute-error
+    if len(err) == 0:
+        return 'Unknown'
+    code = err.get('Code', 'Unknown')  # pytype: disable=attribute-error
+    if code == 'Unknown':
+        code = err.get('code', 'Unknown')  # pytype: disable=attribute-error
```

Comment:
If your patch for this func can return 499, you only need update `s3_should_retry` under the below. `client._make_request` is patched for retrying request, `client._make_request` is invoked by `client.put_object` and `client.upload_part` .

Reply:
Add 499 to s3_should_retry.